src/main/java/org/spongepowered/common/util/gen/ArrayMutableBlockBuffer.java

```diff
@@ -66,16 +70,10 @@ public ArrayMutableBlockBuffer(BlockPalette palette, Vector3i start, Vector3i si
         this.data = new CharBackingData(blocks);
     }
 
-    public ArrayMutableBlockBuffer(BlockPalette palette, Vector3i start, Vector3i size, int[] blocks) {
-        super(start, size);
-        this.palette = palette;
-        this.data = new IntBackingData(blocks);
-    }
-
     ArrayMutableBlockBuffer(BlockPalette palette, Vector3i start, Vector3i size, BackingData blocks) {
         super(start, size);
         this.palette = palette;
-        this.data = blocks.copyOf();
```

Comment:
O that one... There is only one usage, and thats [here](https://github.com/xcube16/SpongeCommon/blob/7621cc4a8028c385830ecfcd8a10350434c41e10/src/main/java/org/spongepowered/common/util/gen/ArrayMutableBlockBuffer.java#L132).
I like to think about it like new String(char[]), string being immutable, makes a copy of the char[] array so it cant be mutated form the outside.
However its ok if ArrayMutableBlockBuffer is changed from the outside using the char[].

This does get me thinking though... ArrayImmutableBlockBuffer() takes a palette argument. What if BlockPalette#remove() was called after ArrayImmutableBlockBuffer() was created. (Maybe this can be fixed in another PR)